warehouse/query-core/src/main/java/datawave/query/jexl/lookups/ExpandedFieldCache.java

```diff
@@ -0,0 +1,54 @@
+package datawave.query.jexl.lookups;
+
+import java.util.Map;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ * A cache for actual values retrieved from the index when an unfielded lookup occurs. Keeps track of whether those field / term pairs have previously been
+ * expanded.
+ */
+public class ExpandedFieldCache {
+    private Multimap<String,ValueSet> previouslyExpandedFieldCache = HashMultimap.create();
+
+    private boolean containsExpansionsFor(IndexLookupMap fieldstoterms) {
+        if (!fieldstoterms.isKeyThresholdExceeded()) {
+            for (Map.Entry<String,ValueSet> fieldTermPair : fieldstoterms.entrySet()) {
+                if (!fieldTermPair.getValue().isThresholdExceeded()) {
+                    if (previouslyExpandedFieldCache.containsEntry(fieldTermPair.getKey(), fieldTermPair.getValue())) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean containsExpansionsFor(String fieldName, String literal) {
+        if (previouslyExpandedFieldCache.containsKey(fieldName)) {
+            for (ValueSet value : previouslyExpandedFieldCache.get(fieldName)) {
+                if (value.contains(literal)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public void addExpansion(IndexLookupMap fieldstoterms) {
+        if (!containsExpansionsFor(fieldstoterms)) {
```

Comment:
This does not look correct to me.  The test only tests of one of the terms actually exists in the cache.  So if only one of them exists then it will not add any of the fieldstoterms?  I dont think the test is desired here.